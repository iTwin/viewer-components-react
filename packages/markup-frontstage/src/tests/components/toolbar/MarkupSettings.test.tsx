import { Point, UiCore } from "@bentley/ui-core";
import { render } from "@testing-library/react";
import React from "react";
// import { MockService } from "../../util/MockService";

// import MarkupSettingsPanel from "../../../components/toolbar/MarkupSettings";
// import { I18N } from "@bentley/imodeljs-i18n";

// jest.mock("@bentley/ui-abstract");
// jest.mock("@bentley/ui-components");
// jest.mock("@bentley/ui-core");
// jest.mock("@bentley/ui-framework");
// jest.mock("@bentley/imodeljs-frontend");

describe("MarkupSettings tool testing", () => {
  // it("should render", async () => {
  //   //arrange
  //   UiCore.initialize(new I18N());
  //   const { queryByTestId } = render(
  //     <MarkupSettingsPanel point={Point.create({ x: 0, y: 0 })} />
  //   );

  //   //act
  //   const settingPanel = queryByTestId("markup-settings-tool");

  //   // //assert
  //   // expect(settingPanel).toMatchSnapshot();

  //   expect(true).toBeTruthy();
  // });

  it("test truthy", () => {
    expect(true).toBeTruthy();
  });
});
