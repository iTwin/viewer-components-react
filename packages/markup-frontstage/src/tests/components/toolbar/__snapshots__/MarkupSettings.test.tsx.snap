// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkupSettings tool testing should render 1`] = `
<div
  data-testid="markup-settings-tool"
>
  <div
    class="uifw-position-popup"
    style="top: 0px; left: 0px;"
  >
    <div
      class="markupContainer"
    >
      <div
        class="markupContainerSetting"
        data-testid="stroke-color-picker-button"
      >
        <span
          class="toolName"
        />
        <button
          class="components-colorpicker-button"
          data-testid="components-colorpicker-button"
        />
      </div>
      <div
        class="markupContainerSetting"
        data-testid="fill-color-picker-button"
      >
        <span
          class="toolName"
        />
        <button
          class="components-colorpicker-button"
          data-testid="components-colorpicker-button"
        />
      </div>
      <div
        class="markupContainerSetting"
        data-testid="stroke-weight-picker-button"
      >
        <span
          class="toolName"
        />
        <div>
          <button
            aria-expanded="false"
            aria-haspopup="listbox"
            class="components-lineweight-swatch components-weightpicker-button"
            data-testid="components-weightpicker-button"
          >
            <span>
              3
            </span>
            <div
              style="height: 3px;"
            />
          </button>
        </div>
      </div>
      <div
        class="markupContainerSetting"
        data-testid="alpha-slider"
      >
        <span
          class="toolName"
        />
        <div
          class="components-alpha-container-horizontal markupTransparencySlider"
          data-testid="alpha-container"
        >
          <div
            aria-label="Transparency"
            aria-valuemax="1"
            aria-valuemin="0"
            aria-valuenow="0.8"
            class="components-alpha-slider"
            data-testid="alpha-slider"
            role="slider"
            tabindex="0"
          >
            <div
              class="components-alpha-pointer"
              data-testid="alpha-pointer"
              style="left: 80%;"
            />
          </div>
        </div>
      </div>
      <div
        class="markupContainerSetting"
        data-testid="text-color-picker-button"
      >
        <span
          class="toolName"
        />
        <button
          class="components-colorpicker-button"
          data-testid="components-colorpicker-button"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`MarkupSettings tool testing should render color picker dropdown when clicked on color picking button 1`] = `
<div
  class="components-colorpicker-popup-colors"
  data-testid="components-colorpicker-popup-colors"
  style="grid-template-columns: repeat(4, 1fr);"
>
  <button
    class="components-color-swatch components-colorpicker-swatch"
  />
  <button
    class="components-color-swatch components-colorpicker-swatch"
  />
  <button
    class="components-color-swatch components-colorpicker-swatch"
  />
  <button
    class="components-color-swatch components-colorpicker-swatch"
  />
  <button
    class="components-color-swatch components-colorpicker-swatch"
  />
  <button
    class="components-color-swatch components-colorpicker-swatch"
  />
  <button
    class="components-color-swatch components-colorpicker-swatch"
  />
  <button
    class="components-color-swatch components-colorpicker-swatch"
  />
  <button
    class="components-color-swatch components-colorpicker-swatch"
  />
  <button
    class="components-color-swatch components-colorpicker-swatch"
  />
  <button
    class="components-color-swatch components-colorpicker-swatch"
  />
  <button
    class="components-color-swatch components-colorpicker-swatch"
  />
  <button
    class="components-color-swatch components-colorpicker-swatch"
  />
  <button
    class="components-color-swatch components-colorpicker-swatch"
  />
  <button
    class="components-color-swatch components-colorpicker-swatch"
  />
  <button
    class="components-color-swatch components-colorpicker-swatch"
  />
</div>
`;

exports[`MarkupSettings tool testing should render weight picker dropdown when clicked on weight picking button 1`] = `
<ul
  class="components-weightpicker-popup-lines"
  data-testid="components-weightpicker-popup-lines"
>
  <button
    class="components-lineweight-swatch components-weightpicker-swatch"
    id="ui-core-lineweight-1"
  >
    <span>
      1
    </span>
    <div
      style="height: 1px;"
    />
  </button>
  <button
    class="components-lineweight-swatch components-weightpicker-swatch"
    id="ui-core-lineweight-2"
  >
    <span>
      2
    </span>
    <div
      style="height: 2px;"
    />
  </button>
  <button
    class="components-lineweight-swatch components-weightpicker-swatch active"
    id="ui-core-lineweight-3"
  >
    <span>
      3
    </span>
    <div
      style="height: 3px;"
    />
  </button>
  <button
    class="components-lineweight-swatch components-weightpicker-swatch"
    id="ui-core-lineweight-4"
  >
    <span>
      4
    </span>
    <div
      style="height: 4px;"
    />
  </button>
  <button
    class="components-lineweight-swatch components-weightpicker-swatch"
    id="ui-core-lineweight-5"
  >
    <span>
      5
    </span>
    <div
      style="height: 5px;"
    />
  </button>
  <button
    class="components-lineweight-swatch components-weightpicker-swatch"
    id="ui-core-lineweight-6"
  >
    <span>
      6
    </span>
    <div
      style="height: 6px;"
    />
  </button>
  <button
    class="components-lineweight-swatch components-weightpicker-swatch"
    id="ui-core-lineweight-7"
  >
    <span>
      7
    </span>
    <div
      style="height: 7px;"
    />
  </button>
  <button
    class="components-lineweight-swatch components-weightpicker-swatch"
    id="ui-core-lineweight-8"
  >
    <span>
      8
    </span>
    <div
      style="height: 8px;"
    />
  </button>
  <button
    class="components-lineweight-swatch components-weightpicker-swatch"
    id="ui-core-lineweight-9"
  >
    <span>
      9
    </span>
    <div
      style="height: 9px;"
    />
  </button>
  <button
    class="components-lineweight-swatch components-weightpicker-swatch"
    id="ui-core-lineweight-10"
  >
    <span>
      10
    </span>
    <div
      style="height: 10px;"
    />
  </button>
</ul>
`;
